client/src/components/molecules/OngoingAgendaCard.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react";
+import { Box, Card, Text, Button } from "@/components/atoms";
+import { OngoingAgenda } from "biseo-interface/agenda";
+import {
+  ChoiceComponent,
+  CompletedChoice,
+  NotVotableChoice,
+} from "@/components/molecules";
+import { useAgenda } from "@/services/agenda";
+
+type OngoingAgendaProps = {
+  agenda: OngoingAgenda;
+};
+
+export const OngoingAgendaCard: React.FC<OngoingAgendaProps> = ({ agenda }) => {
+  const [chosenChoiceId, setChosenChoiceId] = useState(0);
+  const { voteAgenda } = useAgenda(state => ({
+    voteAgenda: state.voteAgenda,
+  }));
+
+  const vote = () => {
+    if (chosenChoiceId === 0) {
+      return null;
+    } else {
+      voteAgenda(chosenChoiceId, agenda.id);
+    }
+  };
```

Comment:
이와 같이 쓰이는 함수를 이렇게 작성하면 해당 컴포넌트가 re-render 될 때마다 evaluate 되기 때문에 비효율적입니다. `useCallback` 훅을 통해서 최적화할 수 있습니다.
```typescript
import { ..., useCallback } from "react";
```

```suggestion
  const vote = useCallback(() => {
    if (chosenChoiceId === 0) {
      return null;
    } else {
      voteAgenda(chosenChoiceId, agenda.id);
    }
  }, [voteAgenda, chosenChoiceId, agenda.id]);
```

`useCallback`에 대한 내용은 아래 링크에 잘 설명되어 있으니 참고하시면 될 것 같아요! 앞으로도 아마 `useMemo`와 함께 쓸 일이 많을 거에요.
- https://react.vlpt.us/basic/18-useCallback.html